src/Engine/Scene/SkeletonComponent.hpp

```diff
@@ -0,0 +1,243 @@
+ *    - the Skeleton;
+ *    - the mapping from the bones RenderObject's index to
+ *      the bones indices within the Skeleton;
+ *    - the reference Pose;
+ *    - the current Animation;
+ *    - the current animation time;
+ *    - whether the animation time has been reset.
+ */
+class RA_ENGINE_API SkeletonComponent : public Component
+{
+  public:
+    friend class SkeletonBasedAnimationSystem;
+
+    /// Animations are lists of keyframed transforms (one per bone).
+    using Animation = std::vector<Core::Animation::KeyFramedValue<Core::Transform>>;
+
+    SkeletonComponent( const std::string& name, Entity* entity );
+    ~SkeletonComponent() override;
+    SkeletonComponent( const SkeletonComponent& ) = delete;
+    SkeletonComponent& operator=( const SkeletonComponent& ) = delete;
+
+    /// \name Component interface
+    /// \{
+
+    void initialize() override {}
+
+    bool canEdit( const Core::Utils::Index& roIdx ) const override;
+
+    Core::Transform getTransform( const Core::Utils::Index& roIdx ) const override;
+
+    void setTransform( const Core::Utils::Index& roIdx, const Core::Transform& transform ) override;
+    /// \}
+
+    /// \name Build from fileData
+    /// \{
+
+    /// Create the skeleton from the given data.
+    void handleSkeletonLoading( const Core::Asset::HandleData* data );
+
+    /// Create the animations from the given data.
+    void handleAnimationLoading( const std::vector<Core::Asset::AnimationData*>& data );
+    /// \}
+
+    /// \name Skeleton-based animation data
+    /// \{
+
+    /// Returns the name of the skeleton.
+    inline const std::string& getSkelName() const { return m_skelName; }
+
+    /// Set the animation skeleton.
+    void setSkeleton( const Core::Animation::Skeleton& skel );
+
+    /// Returns the animation skeleton.
+    inline const Core::Animation::Skeleton* getSkeleton() { return &m_skel; }
+
+    /// Return the number of animations.
+    inline size_t getAnimationCount() const { return m_animations.size(); }
+
+    /// Return the \p i -th animation.
+    inline const Animation& getAnimation( const size_t i ) const { return m_animations[i]; }
+
+    /// Return the \p i -th animation.
+    inline Animation& getAnimation( const size_t i ) { return m_animations[i]; }
+
+    /// Creates a new empty animation from the current pose.
+    Animation& addNewAnimation();
+
+    /// Deletes the \p i-th animation.
+    void removeAnimation( const size_t i );
+
+    /// Set the animation to play.
+    void useAnimation( const size_t i );
+
+    /// Return the index of the animation to play.
+    size_t getAnimationId() const;
+    /// \}
+
+    /// \name Animation Process
+    /// \{
+
+    /// Updates the skeleton pose as the pose corresponding to time \p time.
+    void update( Scalar time );
+
+    /// Returns the current time of animation.
+    Scalar getAnimationTime() const;
+
+    /// Returns the duration of the current animation.
+    Scalar getAnimationDuration() const;
```

Comment:
Since there is `getAnimationTimeInterval`, I think we should get rid of this one (which is not used anymore b.t.w.)

Reply:
We can keep it for now as there is no impact, no cost, and it can be used in several uses cases.

Reply:
well, one just have to to the diff between the time interval bounds...
Plus it's a remnant from the old way to manage time in the animation system